wavespectra/input/__init__.py

```diff
@@ -64,12 +64,18 @@ def open_netcdf_or_zarr(filename_or_fileglob, file_format, mapping={}, chunks={}
         - dset (Dataset): spectra dataset object read from ww3 file.
 
     """
-    # Allow chunking using wavespctra names
+    # Allow chunking using wavespectra names
     _chunks = chunks_dict(chunks, mapping)
     if file_format == "netcdf":
-        dset = xr.open_mfdataset(
-            filename_or_fileglob, chunks=_chunks, combine="by_coords"
-        )
+        # Use mfdataset to open if a filename or fileglob is passed
+        # otherwise we assume a fileobj is passed and open it directly with
+        # xr.open_dataset
+        try:
+            dset = xr.open_mfdataset(
+                filename_or_fileglob, chunks=_chunks, combine="by_coords"
+            )
+        except ValueError:
```

Comment:
I get an `IndexError` rather than a `ValueError` when trying to open a file-like object with xr.open_mfdataset - do you get `ValueError` in wavespectra?

Reply:
This gives me a `ValueError`, saying that it does not know which engine to use because it is just a file like object passed in.

```python
import xarray as xr

with open('./sample_files/swanfile.nc', 'rb') as file:
    xr.open_mfdataset(file, chunks={}, combine="by_coords")
```

If I specify the engine, that seems to give me a `ValueError` as well: 

```python
with open('./sample_files/swanfile.nc', 'rb') as file:
    xr.open_mfdataset(file, chunks={}, combine="by_coords", engine="netcdf4")
```

gives:

```
ValueError: can only read bytes or file-like objects with engine='scipy' or 'h5netcdf'
```

Switching to `h5netcdf` also throws a `ValueError` for me: 

```
ValueError: can't open netCDF4/HDF5 as bytes try passing a path or file-like object
```

Which doesnt totally make sense, but throwing an error does. I am unable to get an `IndexError`, can you show me the code to reproduce the `IndexError`? I'm also happy to just catch any exception, but better to be explicit